docs/introduction/faq.md

```diff
@@ -0,0 +1,90 @@
+Produkty je možné rozdělit do několika skupin podle toho, jak se s těmito produkty pracuje nebo k čemu jsou používaný.
+
+**Visible** - produkty, které jsou v databázi vedeny jako viditelné pro konkrétní cenovou skupinu a konkrétní doménu.
+Produkt totiž muže být viditelný na jedné doméně ale skrytý na druhé doméně, protože nemá např pro druhou doménu nastavenou cenu.
+Současně, když je produkt viditelný, to ještě neznamená, že je možné tento produkt i zakoupit.
+
+**Offered** - produkty, které splňují podmínky pro **visible** a současně jsou v databázi vedeny s příznakem `calculatedSellingDenied` = `FALSE`.
+Příznak `calculatedSellingDenied` určuje, zdali není náhodou produkt vyprodán nebo v případě varianty, zdali není zakázaný prodej hlavní rodičovské položky, pod kterou táto varianta patří.
+
+**Listable** - produkty, které splňuji podmínky pro **offered** a současně se nejedná o konkrétní varianty nějaké hlavní varianty.
+Pro zobrazení ve výpisech produktu totiž nezobrazujeme konkrétní varianty položky ale zobrazujeme jen hlavní rodičovskou variantu dané konkrétní varianty.
+
+**Sellable** - produkty, které splňuji podmínky pro **offered** a současně se nejedná o hlavní variantu (muže se tedy jednat o konkrétní variantu, nebo stadartní položku).
+Sellable určuje položku, kterou je možné zakoupit.
+Zakoupit lze jen standartní položky a konkrétní varianty nějaké hlavní rodičovské varianty.
+
+## How calculated attributes work?
+Na platformě Shopsys Framework existuje několik atributu, které se nanastavují přímo ale jejich hodnota je výsledkem nějakých automatických přepočtu atributu, např viditelnost oddělení, calculated visibility of product, etc.
+Tato vypočtená hodnota je závislá na dalších parametrech daného objektu, např oddělení, které nemá pro locale konkrétní domény vyplněno název, bude na této doméně nastaveno jako neviditelné.
+Samotné přepočty atributu mužou být inicializovány jako `immediate` nebo `scheduled`:
+
+**immediate** - přepočet daného atributu je spuštěný při vyvolání události `kernel.response`.
+See například třídu `Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPriceRecalculator` and method `onKernelResponse`.
+
+**scheduled** - přepočet atributu je spuštěn později.
+Produkt nebo oddělení je jen označen k pozdejšímu přepočtu a samotný přepočet proběhne později - po spuštění přislušného cron modulu, see `Shopsys\FrameworkBundle\Command\RecalculationsCommand`.
+
+Např v metodě `edit` v `Shopsys\FrameworkBundle\Model\Product\ProductFacade` se volá metoda `scheduleProductForImmediateRecalculation` třídy `Shopsys\FrameworkBundle\Model\Product\Availability\ProductAvailabilityRecalculationScheduler`.
+Toto volání zabezpečí, že při volání metody `edit` je produkt označen k `immediate` přepočtu dostupnosti.
+Tento přepočet je následně spuštěn po dokončení requestu a odchycení metody `onKernelResponse`.
+Neplatí tedy fakt, že po provedení metody `edit` má již produkt dostupnost dopočtenou.
```

Comment:
Ok, I can add this information. But Im little confused because I have never seen such a explanation in  our docs. There is only information WHAT and HOW, not WHY .

Reply:
And truth is, this is a legacy feature. In the past, recalculators were based on the raw queries. Later there were create Recalculators classes and this functionality was moved into objects. But the usage of kernel.response event is still there and Im sure it will be removed in the future. And I dont want to see this "storyline" in the FAQ section.

Reply:
In different words you say: the current solution is wrong and should be changed? If you are right, create issue suggesting this changes and mention here, this is "legacy" and there is the intention to change it in the future and maybe link it with the issue.

Reply:
Ok, right now im confused for sure....

1. Should I create the issue? YES/NO

2. Should I mention the legacy code in the FAQ section? YES/NO

3. Should I add the link to the new issue? YES/NO

Reply:
1, 2, 3 YES but the decision is yours. I was not aware, that it is legacy, but if you are sure, it is then it is best to mention and get rid of it, right?

Reply:
Done